program/src/instructions/compressed/buy_spl.rs

```diff
@@ -179,7 +196,9 @@ pub fn process_buy_spl<'info>(
 ) -> Result<()> {
     let list_state = &ctx.accounts.list_state;
 
+    // TODO: do we even do anything with this v? what's the point of it?
```

Comment:
Old IDLs might pass in a creator in the cosigner slot whereas new IDL passes in an optional cosigner (cosigner or tcomp ID to represent `None`). The algorithm to handle this is:

```
Get cosigner
Case 1: New IDL Option<Cosigner>
Is key a signer? (Any key that is a signer) --> Some(cosigner)
Is key crate ID? TCMPhJdwDryooaGtiocG1u3xcYbRpiJzb283XfCZsDp --> None
Case 2: Old IDL
Above cases are false, this is the first creator
```